daemons/ipa-otpd/oauth2.c

```diff
@@ -426,7 +428,7 @@ int oauth2(struct otpd_queue_item **item, enum oauth2_state oauth2_state)
     char *args[50] = {NULL};
     size_t args_idx = 0;
     krb5_data data_state = {0};
-    struct otpd_queue_item *saved_item;
+    struct otpd_queue_item *saved_item = NULL;
```

Comment:
Is it expected by the other code?
For example, `oauth2_on_child_writable` uses `saved_item` without `NULL` check, though I don't know whether that code is reachable or not.

Reply:
There are two states processed by those callbacks: get device code and get access token. `saved_item` is only used in the case of getting the access token. In `oauth2_on_child_writable` we short-circuit the get device code state and never reach the place where `saved_item` is used.

Reply:
Thank you for clarification, LGTM then.